apps/ecommerce/frontend/src/components/Common/ProductCard/ProductCardBadge/ProductCardBadge.tsx

```diff
@@ -0,0 +1,37 @@
+import { Badge, Box, EntityStatus } from '@contentful/f36-components';
+import { ExternalResource, ExternalResourceError } from 'types';
+import { styles } from './ProductCardBadge.styles';
+
+interface ProductCardBadgeProps {
+  showHeaderMenu?: boolean;
```

Comment:
I think it's somewhat of an anti-pattern to have these simple badge components handle all this optional/nullable logic. There should be instead be multiple types of ProductCardBadges, one for the good state, and the other for missing states and let the parent handle which one to render. That way these simpler badge components are more re-usable

Reply:
yeah this is something I certainly debated and is a good callout. The reason I decided to move the logic to this component itself was simply because the atom it is creating is already existent within F36, and any customization added is a very specific implementation of the component (limiting usability). Otherwise would certainly opt for just moving the logic to the parent component. Its kind of a fine line imo, congesting the parent with many different conditional renders, or pulling those out into a hook or some helper, or just adding simple logic to the child component if it is a very specific implementation, which imo this is a very specific implementation of the Badge and I somewhat foresee us adding more conditionals given the future differing states of a ProductCard. In general I think parents should handle conditional rendering until you reach several different variations, which is what I am noticing here, so I treat this more as an adjacent component handling logic, instead of a lower level atom. But that was just my original rationale and totally agree with your point 👍

Reply:
Basically you would pass in the Badge from the parent, and then the HOC component would return the modified Badge with whatever wrapper components around it.

Honestly, I don't think this optimization matters - I think there's a very good chance that the Platform team will re-design/build on top and change a lot of this. So I'll let them decide what to do here!

Reply:
ah okay I see, I like the idea, but agreed that things will be in flux with this!